roles/ceph-defaults/tasks/facts.yml

```diff
@@ -221,15 +221,20 @@
 - block:
     - name: get current cluster status (if already running)
       command: "{{ docker_exec_cmd }} ceph --cluster {{ cluster }} -s -f json"
+      changed_when: false
+      failed_when: false
+      check_mode: no
       register: ceph_current_status
 
     - name: set_fact ceph_current_status (convert to json)
       set_fact:
         ceph_current_status: "{{ ceph_current_status.stdout | from_json }}"
+      when: ceph_current_status.rc == 0
 
     - name: set_fact rgw_hostname
       set_fact:
         rgw_hostname: "{% for key in ceph_current_status['servicemap']['services']['rgw']['daemons'].keys() %}{% if key == ansible_fqdn %}{{ key }}{% endif %}{% endfor %}"
+      when: ceph_current_status.rc == 0
   when:
     - ceph_current_fsid.get('rc', 1) == 0
```

Comment:
This task should prevent us from entering the block if no cluster is running, so it's still unclear why you ran into this.

Reply:
This is the output of the task that is supposed to set `ceph_current_fsid`. But the task is delegated to `"{{ groups[mon_group_name][0] }}"` and that's why it's returning `rc=0`.

```
ok: [aio1_ceph-rgw_container-9474601c -> 172.29.238.208] => {
    "changed": false, 
    "cmd": [
        "timeout", 
        "5", 
        "ceph", 
        "--cluster", 
        "ceph", 
        "fsid"
    ], 
    "delta": "0:00:00.524653", 
    "end": "2018-08-15 08:23:15.892404", 
    "failed_when_result": false, 
    "invocation": {
        "module_args": {
            "_raw_params": "timeout 5  ceph --cluster ceph fsid", 
            "_uses_shell": false, 
            "chdir": null, 
            "creates": null, 
            "executable": null, 
            "removes": null, 
            "stdin": null, 
            "warn": true
        }
    }, 
    "rc": 0, 
    "start": "2018-08-15 08:23:15.367751", 
    "stderr": "", 
    "stderr_lines": [], 
    "stdout": "391701ff-eaf3-4b93-91f6-4b2f57937f74", 
    "stdout_lines": [
        "391701ff-eaf3-4b93-91f6-4b2f57937f74"
    ]
}
```

However, the problematic task in this PR is not delegated as you said and it is running on the wrong node.